### PR TITLE
Add Example Files

### PR DESCRIPTION
**Pull request by gomesjason**

### CODE REVIEW

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Dynamic Provisioning
+
+## Context
+
+FSx OpenZFS resources can be mounted to a give pod either statically or dynamically.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: FSx for OpenZFS* given*

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Static provisioning requires the user to initially create the FSxZ resource manually.
+Dynamic provisioning will automatically create FSxZ resources based on user specifications.
+
+This guide will detail steps needed to dynamically create and mount a FSxZ resource.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: an* FSxZ resource, this applies elsewhere

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+For details on statically mounting a FSxZ resource see this [guide](../static-provisioning/README.md).
+
+This CSI driver supports the use of both FSxZ filesystems and FSxZ volumes as a container storage interfaces.
+This guide will detail steps needed to deploy both types of resources.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: detail the steps*

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+See this [guide](https://docs.aws.amazon.com/fsx/latest/OpenZFSGuide/administering-file-systems.html) for more details on each.
+
+## Prerequisites
+
+1. Kubernetes 1.13+ (CSI 1.0).
+2. The [aws-fsx-openzfs-csi-driver](https://github.com/kubernetes-sigs/aws-fsx-openzfs-csi-driver) installed.
+
+## Usage
+
+This example shows you how to create, and mount a FSxZ filesystem and FSxZ volume dynamically.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: unnecessary comma, I think it reads better if we just say "This example shows you how to dynamically provision an FSxZ file system and FSxZ volume in your cluster" (or something along those lines)

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Values in the example files may be modified or removed based on preferences.
+See this [document](../../../docs/parameters.md) for parameter configurations.
+
+### Create and Mount a FSxZ Resource
+
+1. Run **one** following to apply manifests needed to create either a FSxZ filesystem or a FSxZ volume.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: one of the following*
to apply the manifests necessary*

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    1. FSxZ filesystem:
+    ```sh
+   kubectl apply -f filesystem/,manifests/
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md:32]: Nice! I like this notation, didn't know you could use commas to separate them out like this.

File: examples/kubernetes/dynamic-provisioning/filesystem/storageclass.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+parameters:
+  volumeType: "filesystem"
+  kmsKey: "12345678-90ab-cdef-ghij-klmnopqrstuv"
+  automaticBackupRetentionDays: "1"
+  copyTagsToBackups: "false"
+  copyTagsToVolumes: "false"
+  dailyAutomaticBackupStartTime: "00:00"
+  deploymentType: "SINGLE_AZ_1"
+  diskIopsConfiguration: "Iops=300,Mode=USER_PROVISIONED"
+  rootVolumeConfiguration: "CopyTagsToSnapshots=true,DataCompressionType=NONE,NfsExports=[{ClientConfigurations=[{Clients=*,Options=[rw,crossmnt]}]}],ReadOnly=false,RecordSizeKiB=128,UserAndGroupQuotas=[{Id=1,StorageCapacityQuotaGiB=10,Type=USER}]"
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/filesystem/storageclass.yaml:15]: Can you add comments indicating which parameters are required?

File: examples/kubernetes/dynamic-provisioning/volume/storageclass.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fsx-sc
+provisioner: fsx.openzfs.csi.aws.com
+parameters:
+  volumeType: "volume"
+  copyTagsToSnapshots: "false"
+  dataCompressionType: "NONE"
+  nfsExports: "[{ClientConfigurations=[{Clients=*,Options=[rw,crossmnt]}]}]"
+  parentVolumeId: "fsvol-03062e7ff37662dff"
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/volume/storageclass.yaml]: same here, add comments for volumeType and parentVolumeId to indicate that they're required

File: examples/kubernetes/snapshot/README.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
+# Snapshots
+
+## Context
+
+Volume snapshots allows users to quickly store the state of volume at a given time.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md]: Volume snapshots allow*
state of a* volume

File: examples/kubernetes/snapshot/README.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+Snapshots can be taken either manually, or utilizing the CSI driver.
+Any snapshot taken can be utilized to create new volumes.
+
+Currently only FSxZ volume level snapshots are supported. 
+FSxZ filesystem level backups are NOT supported.
+See the [design document](../../../docs/design.md) for more information.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md]: The design document doesn't exist. Do you plan on adding one in the future? I'm not convinced it's necessary.

File: examples/kubernetes/snapshot/README.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
+# Snapshots
+
+## Context
+
+Volume snapshots allows users to quickly store the state of volume at a given time.
+Snapshots can be taken either manually, or utilizing the CSI driver.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md]: either manually or by utilizing the CSI driver*

File: examples/kubernetes/snapshot/README.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+Any snapshot taken can be utilized to create new volumes.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md:7]: I'd add that we can only create a new volume from a snapshot if it exists under the same file system. That is a current limitation of our ZFS snap-send-rcv feature.

File: examples/kubernetes/snapshot/README.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+2. The [aws-fsx-openzfs-csi-driver](https://github.com/kubernetes-sigs/aws-fsx-openzfs-csi-driver) installed.
+3. The [external snapshotter](https://github.com/kubernetes-csi/external-snapshotter) installed.
+4. The `VolumeSnapshotDataSource` is set in `--feature-gates=` in the `kube-apiserver` specification. This feature is enabled by default from Kubernetes v1.17+.
+
+## Usage
+
+This example shows you how to create snapshots and restore them.
+
+Values in the example files may be modified or removed based on preferences.
+See this [document](../../../docs/parameters.md) for parameter configurations.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md]: This file doesn't exist yet, but we should add it eventually. Were you planning on adding it later?

File: examples/kubernetes/snapshot/README.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+    ```sh
+    kubectl describe volumesnapshot.snapshot.storage.k8s.io fsx-vs
+    ```
+3. Delete the created Kubernetes resources:
+    ```sh
+    kubectl delete -f existing-snapshot/
+    ```
+
+### Create VolumeSnapshot From PersistentVolume
+1. Run **one** following to apply manifests needed to create either a FSxZ filesystem or a FSxZ volume.
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/README.md]: one of the following*

File: examples/kubernetes/snapshot/snapshot/class.yaml [hunk 15]
```diff
@@ -0,0 +1,8 @@
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshotClass
+metadata:
+  name: fsx-snapshot-class
+driver: fsx.openzfs.csi.aws
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/snapshot/class.yaml]: fsx.openzfs.csi.aws.com*

File: examples/kubernetes/snapshot/existing-snapshot/content.yaml [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+  name: fsx-snapshot-content
+spec:
+  volumeSnapshotRef:
+    kind: VolumeSnapshot
+    name: snapshot
+    namespace: default
+  source:
+    snapshotHandle: fsvolsnap-0809d8f1df40a7427
+  driver: fsx.openzfs.csi.aws.com
+  deletionPolicy: Delete
```
Comment (hughdanliu) [inline examples/kubernetes/snapshot/existing-snapshot/content.yaml]: nit: extra lines at the end of files

File: examples/kubernetes/static-provisioning/README.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
+# Static Provisioning
+
+## Context
+
+FSxZ resources can be mounted to a give pod either statically or dynamically.
```
Comment (hughdanliu) [inline examples/kubernetes/static-provisioning/README.md]: Not going to dive super deep into this one, but most of the comments left on the README for dynamic provisioning are also applicable here.

File: examples/kubernetes/static-provisioning/volume/pv.yaml [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+  accessModes:
+    - ReadWriteMany
+  mountOptions:
+    - nfsvers=4.1
+    - rsize=1048576
+    - wsize=1048576
+    - timeo=600
+  persistentVolumeReclaimPolicy: Delete
+  csi:
+    driver: fsx.openzfs.csi.aws
```
Comment (hughdanliu) [inline examples/kubernetes/static-provisioning/volume/pv.yaml]: fsx.openzfs.csi.aws.com*

File: examples/kubernetes/volume-expansion/README.md [hunk 19]
```diff
@@ -0,0 +1,50 @@
+# Volume Expansion
+
+## Context
+
+Users are able to change the capacity of their volumes. 
+Currently, the capacity can ONLY be increased, and NOT decreased.
+Only storage capacity can be adjusted. Throughput can NOT be modified.
+See the [design document](../../../docs/design.md) for more information.
```
Comment (hughdanliu) [inline examples/kubernetes/volume-expansion/README.md]: no design document yet, also I'd add a disclaimer that scaling the capacity on an FSxZ volume increases the StorageCapacityQuota and StorageCapacityReservation to the same value (the requested storage)

File: examples/kubernetes/dynamic-provisioning/README.md [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+## Context
+
+FSx OpenZFS resources can be mounted to a give pod either statically or dynamically.
+Static provisioning requires the user to initially create the FSxZ resource manually.
+Dynamic provisioning will automatically create FSxZ resources based on user specifications.
+
+This guide will detail steps needed to dynamically create and mount a FSxZ resource.
+For details on statically mounting a FSxZ resource see this [guide](../static-provisioning/README.md).
+
+This CSI driver supports the use of both FSxZ filesystems and FSxZ volumes as a container storage interfaces.
```
Comment (hughdanliu) [inline examples/kubernetes/dynamic-provisioning/README.md]: as container storage interfaces*

File: examples/kubernetes/snapshot/snapshot/class.yaml (hunk 15 above)
Comment (gomesjason) [reply examples/kubernetes/snapshot/snapshot/class.yaml]: Good catch. You saved me some headache when testing this morning.

File: examples/kubernetes/snapshot/README.md (hunk 10 above)
Comment (gomesjason) [reply examples/kubernetes/snapshot/README.md]: I planned on adding a design doc along with a parameters doc. The design doc would point out specific decisions and limitations such as throughput not being expandable, and why SkipFinalBackups is required. I'm not opposed to doing a FAQ instead although.

File: examples/kubernetes/snapshot/README.md (hunk 13 above)
Comment (gomesjason) [reply examples/kubernetes/snapshot/README.md]: Yea I plan on adding it later when I do the documents task.

**Pull request closed without merging**